bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/util/NLS.java

```diff
@@ -141,18 +145,50 @@ public static String bind(String message, Object[] bindings) {
 	 *
 	 * @param baseName the base name of a fully qualified message properties file.
 	 * @param clazz the class where the constants will exist
+	 * @param charset the charset for the given properties file
+	 * @since 3.22
 	 */
-	public static void initializeMessages(final String baseName, final Class<?> clazz) {
+	public static void initializeMessages(final String baseName, final Class<?> clazz, final Charset charset) {
 		if (System.getSecurityManager() == null) {
-			load(baseName, clazz);
+			load(baseName, clazz, charset);
 			return;
 		}
 		AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
-			load(baseName, clazz);
+			load(baseName, clazz, charset);
 			return null;
 		});
 	}
 
+	/**
+	 * Initialize the given class with the values from the message properties specified by the
+	 * base name.  The base name specifies a fully qualified base name to a message properties file,
+	 * including the package where the message properties file is located.  The class loader of the
+	 * specified class will be used to load the message properties resources. Values from the file
+	 * are decoded into characters using the {@link StandardCharsets#UTF_8 UTF-8} {@link Charset charset}.
+	 * <p> This method works as if invoking it were equivalent to evaluating the
+	 * expression:
+	 * <blockquote>{@link
+	 * #initializeMessages(String, Class, Charset)
+	 * NLS.initializeMessages(path, clazz, StandardCharsets.UTF_8)
+	 * }</blockquote>
+	 * <p>
+	 * For example, if the locale is set to en_US and <code>org.eclipse.example.nls.messages</code>
+	 * is used as the base name then the following resources will be searched using the class
+	 * loader of the specified class:
+	 * </p>
+	 * <pre>
+	 *   org/eclipse/example/nls/messages_en_US.properties
+	 *   org/eclipse/example/nls/messages_en.properties
+	 *   org/eclipse/example/nls/messages.properties
+	 * </pre>
+	 *
+	 * @param baseName the base name of a fully qualified message properties file.
+	 * @param clazz the class where the constants will exist
+	 */
+	public static void initializeMessages(final String baseName, final Class<?> clazz) {
+		initializeMessages(baseName, clazz, StandardCharsets.UTF_8);
```

Comment:
I'm not sure UTF-8 is a good default here, as it will most likely break a lot of exiting behavior.

Reply:
This PR's whole purpose is to make UTF-8 the default. I tested it locally against various languages; the behavior is the same against both ASCII-only messages and messages containing escape sequences (this includes anything outside the ASCII range). Please see my original comment on this PR.

Reply:
Please test with a file encoded with `ISO_8859_1` special chars, ASCII is not the default and escaping is not mandatory see [here](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html)

> The [load(InputStream)](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#load-java.io.InputStream-) / [store(OutputStream, String)](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#store-java.io.OutputStream-java.lang.String-) methods work the same way as the load(Reader)/store(Writer, String) pair, except the input/output stream is **encoded in ISO 8859-1 character encoding**. Characters that **cannot be directly represented in this encoding** can be written using Unicode escapes as defined in section 3.3 of The Java™ Language Specification;

This means that it is totally valid to have any ISO 8859-1 in such a file and reading it as UTF-8 will break these existing file.

Reply:
Yes, one must assume that ISO 8859-1 is commonly used.   it seems to me that Java was making some changes in this area and I found this:

https://docs.oracle.com/javase/9/intl/internationalization-enhancements-jdk-9.htm#JSINT-GUID-5ED91AA9-B2E3-4E05-8E99-6A009D2B36AF

![image](https://github.com/user-attachments/assets/368d0e56-ca26-4b49-9b63-608794a753f1)

It's definitely a bad idea to assume UTF-8.